src/sync/app.cpp

```diff
@@ -35,11 +36,30 @@ Optional<T> get_optional(const nlohmann::json& json, const std::string& key)
     return it != json.end() ? Optional<T>(it->get<T>()) : realm::util::none;
 }
 
+static std::map<std::string, std::string> get_request_headers(bool authenticated)
+{
+    std::map<std::string, std::string> headers {
+        { "Content-Type", "application/json;charset=utf-8" },
+        { "Accept", "application/json" }
+    };
+    
+    auto user = SyncManager::shared().get_current_user();
```

Comment:
`if (authenticated && !user)`
in this case, I think it would be appropriate to throw a runtime error? Otherwise this code will silently do the wrong thing.

Reply:
I agree, updated `SyncManager::shared().get_current_user();` has also been moved inside the if block